test/run_test.py

```diff
@@ -0,0 +1,253 @@
+        repo_dir_new = 'RepoDir = /var/tmp'
+        lines = [x if x != repo_dir_old else repo_dir_new for x in lines]
+
+    # change RequireGPGCheck to 0
+    gpg_old = [x for x in lines if x.startswith("RequireGPGCheck")][0]
+    gpg_new = 'RequireGPGCheck = 0'
+    lines = [x if x != gpg_old else gpg_new for x in lines]
+
+    # write new config file
+    with open(new_conf_path, "w") as f:
+        for line in lines:
+            f.write("%s\n" % line)
+
+    # edit env variable RETRACE_SERVER_CONFIG_PATH
+    os.environ["RETRACE_SERVER_CONFIG_PATH"] = new_conf_path
+
+
+# Parse arguments
+parser = argparse.ArgumentParser(description="Run test for retrace-server.")
+parser.add_argument("--coredump", help="path to the coredump, if not \
+                                       specified, will be generated")
+parser.add_argument("--release", help="os_release,\
+                                      default=Fedora release 24 (Twenty Four)")
+parser.add_argument("--executable", help="path to the executable")
+parser.add_argument("--package", help="name of package")
+parser.add_argument("--delete_repo", help="remove repo after bt generation",
+                    action="store_true")
+parser.add_argument("--dont_create_repo", help="don't create repo before bt\
+                    generation", action="store_true")
+parser.add_argument("--interactive", help="run as interactive task, that means\
+                    that task will not be deleted at the end.",
+                    action="store_true")
+args = parser.parse_args()
+
+# set default values
+os_release = args.release if args.release else "Fedora release 24 (Twenty Four)"
+dont_create_repo_arg = True if args.dont_create_repo else False
+delete_repo_arg = True if args.delete_repo else False
+
+# create edited copy of configuration file
+create_local_config(dont_create_repo_arg)
+
+# if coredump specified, parse packages and executable
+if args.coredump:
+    coredump_path = args.coredump
+    package = args.package
+    executable = args.executable
+    if not package or not executable:
+        print ("ERROR: coredump specified, must specify package and executable")
+        sys.exit(1)
+# if not coredump specified, generate it
+else:
+    coredump_path, package, executable = generate_coredump()
+
+# create task
+task = RetraceTask()
+print ("Task ID: "),
+print (task.get_taskid())
+
+if args.interactive:
+    task.set_type(TASK_RETRACE_INTERACTIVE)
+else:
+    task.set_type(TASK_RETRACE)
+task.set("custom_package", package)
+task.set("custom_executable", executable)
+task.set("custom_os_release", os_release)
+
+crashdir = os.path.join(task.get_savedir(), "crash")
+os.mkdir(crashdir)
+# copy coredump to the retrace task directory
+shutil.copy(coredump_path, crashdir)
+
+if not dont_create_repo_arg:
+    corepath = os.path.join(crashdir, "coredump")
+    worker = RetraceWorker(task)
+    # read architecture from coredump
+    arch = worker.read_architecture(None, corepath)
```

Comment:
It would be possible to do that. It also would make sense if only `read_architecture` would be needed. But I need more methods from `RetraceWorker` (such as `read_release_file` or `read_packages`) and these methods are more complicated and are more interconnected to the `RetraceWorker`.